statistics/handle/handle_hist.go

```diff
@@ -353,24 +385,29 @@ func (h *Handle) updateCachedColumn(col model.TableColumnID, colHist *statistics
 func (h *Handle) setWorking(col model.TableColumnID, resultCh chan model.TableColumnID) bool {
 	h.StatsLoad.Lock()
 	defer h.StatsLoad.Unlock()
-	chList, ok := h.StatsLoad.workingColMap[col]
+	chList, ok := h.StatsLoad.WorkingColMap[col]
 	if ok {
-		h.StatsLoad.workingColMap[col] = append(chList, resultCh)
+		if chList[0] == resultCh {
+			return true // just return for duplicate setWorking
```

Comment:
Here is to handle the case when reading column stats failed in the last task and the subLoadWorker tries it again? (Corresponding to the old line 202 `h.StatsLoad.NeededColumnsCh <- task` is deleted) 

So we can make sure that `col` in `WorkingColMap` is set by the current subLoadWorker when handling the last task rather than some other subLoadWorker which might also load `col`'s stats, right?

Reply:
yes